typings/config.d.ts

```diff
@@ -7,29 +7,30 @@
 import Gatherer = require('../lighthouse-core/gather/gatherers/gatherer.js');
 import Audit = require('../lighthouse-core/audits/audit.js');
 
-
 declare global {
   module LH {
     /**
      * The full, normalized Lighthouse Config.
      */
-    export interface Config {
+    export interface Config extends Config.Json {
       settings: Config.Settings;
-      passes?: Config.Pass[];
-      audits?: Config.AuditDefn[];
-      categories?: Record<string, Config.Category>;
-      groups?: Record<string, Config.Group>;
+      passes: Config.Pass[] | null;
+      audits: Config.AuditDefn[] | null;
+      categories: Record<string, Config.Category> | null;
+      groups: Record<string, Config.Group> | null;
     }
 
     module Config {
       /**
        * The pre-normalization Lighthouse Config format.
        */
       export interface Json {
+        extends?: 'lighthouse:default' | 'lighthouse:full' | string | boolean;
```

Comment:
what's the benefit of doing `'lighthouse:default' | string` ?

Reply:
> what's the benefit of doing `'lighthouse:default' | string`?

no benefit, just documentation. I can remove or it might not matter :) (originally it was just the specific strings, but it gets automatically widened to `string` when requiring json files, so default-config et al were failing the type check)